system/mp/mpcomm.cpp

```diff
@@ -1932,16 +1937,29 @@ int CMPConnectThread::run()
         if (sock) {
             try {
                 sock->set_keep_alive(true);
-                size32_t rd;
+                size32_t rd = 0;
                 SocketEndpoint _remoteep;
                 SocketEndpoint hostep;
                 SocketEndpointV4 id[2];
                 traceSlowReadTms("MP: initial accept packet from", sock, &id[0], sizeof(id), sizeof(id), rd, CONFIRM_TIMEOUT, CONFIRM_TIMEOUT_INTERVAL);
-                if (rd != sizeof(id))
+
+                if (0 == rd)
                 {
-                    StringBuffer errMsg("MP Connect Thread: invalid number of connection bytes serialized from ");
+                    if (mpTraceLevel > 1)
+                    {
+                        // cannot get peer addresss as socket state is now ss_shutdown (unless we want to allow this in getPeerEndpoint())
+                        StringBuffer errMsg("MP Connect Thread: connect with no msg received, assumed port monitor check");
+                        PROGLOG("%s", errMsg.str());
+                    }
+                    sock->close();
+                    continue;
+                }
+                else if (rd != sizeof(id))
+                {
+                    // not sure how to get here, it seems either rd == 0 or complete msg is received or an exception occurs
```

Comment:
I think rd can be 0 or rd can be sizeof(id) [complete msg] or an exception could occur so I cannot see how we could ever get to this situation.  This code can probably be deleted but felt I should leave it and make a comment.
I will reword comment.